org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/archetype/ArchetypeGenerator.java

```diff
@@ -145,11 +143,9 @@ private File getEmptyPom(File basedir) {
       try {
         File tempFile = File.createTempFile("pom", ".xml", basedir);
         tempFile.deleteOnExit();
-        try (OutputStreamWriter writer = new OutputStreamWriter(new FileOutputStream(tempFile),
-            StandardCharsets.UTF_8)) {
-          writer.write(
-              "<project><modelVersion>4.0.0</modelVersion><groupId>empty</groupId><artifactId>empty</artifactId><version>1</version><name>Generating archetype</name></project>");
-        }
+        Files.writeString(tempFile.toPath(),
+            "<project><modelVersion>4.0.0</modelVersion><groupId>empty</groupId><artifactId>empty</artifactId><version>1</version><name>Generating archetype</name></project>",
+            StandardCharsets.UTF_8);
```

Comment:
I just noticed that Files.writeString() uses UTF-8 by default so specifying the charset explicitly is actually not necessary.